src/RecordTask.cc

```diff
@@ -1020,6 +1020,19 @@ void RecordTask::record_local(remote_ptr<void> addr, ssize_t num_bytes,
   trace_writer().write_raw(data, num_bytes, addr);
 }
 
+size_t RecordTask::record_remote_by_local_map(remote_ptr<void> addr,
+                                              size_t num_bytes) {
```

Comment:
Yes, that's the intent. This is essentially a helper function. I should probably make it static.